policyenforcer.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratify
+
+import (
+	"context"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// registeredPolicyEnforcers saves the registered policy enforcer factories.
+var registeredPolicyEnforcers map[string]func(config CreatePolicyEnforcerOption) (PolicyEnforcer, error)
+
+// ReferrerValidationReport describes the results of verifying an associated artifact
+// and its nested artifacts by available verifiers.
+type ReferrerValidationReport struct {
+	// Subject is the subject artifact of the current referrer being verified. Required.
+	Subject string
+
+	// Descriptor is the descriptor of the referrer artifact being verified. Required.
+	Descriptor ocispec.Descriptor
+
+	// VerifierReports is reports of verifying current referrer artifact by matching verifiers. Required.
+	VerifierReports []*VerifierResult
+
+	// ReferrerReports is reports of verifying referrer artifacts of current referrer. Required.
+	ReferrerReports []*ReferrerValidationReport
+}
+
+// PolicyEnforcer is an interface with methods that make policy decisions.
+type PolicyEnforcer interface {
+	// EvaluateValidationReports determines the final outcome of validation that
+	// is constructed using the results from individual verifications.
+	EvaluateValidationReports(ctx context.Context, validationReports []*ReferrerValidationReport) bool
+}
+
+// CreatePolicyEnforcerOption represents the options to create a policy enforcer plugin.
+type CreatePolicyEnforcerOption struct {
```

Comment:
code style: make `Options` consistent
```suggestion
type CreatePolicyEnforcerOptions struct {
```